docs/concepts/reference.mdx

```diff
@@ -0,0 +1,135 @@
+
+## Image or Artifact References: Breaking Down the Components
+
+An image reference is essentially the 'address' or 'identifier' you use to
+pull or manage a specific container image or OCI artifact.
+
+There are primarily two contexts in which you'll encounter references:
+
+1. Interacting with a registry to pull or push images.
+2. Interacting with a local OCI Layout store to manage local content.
+
+Let's take a look at the components of an reference in these 2 scenarios.
+
+### What a Reference Means in the Registry
+
+When you pull or push an artifact or image from a registry, you need to specify the reference.
+This reference is a combination of three parts:
+
+1. **Registry**: This is akin to the home or the server where your images or artifacts are stored.
+Think of it as the overarching domain. Examples include Docker Hub (docker.io),
+GitHub Container Registry (ghcr.io), or your own private registry.
+2. **Repository**: This is a segment of the reference, after the registry where a particular 
+application or service's images are stored.
+The repository doesn't necessarily follow a hierarchy within the registry, meaning `a/b/c`
+and `a/b/c/d` are both valid and distinct repositories, with no implied hierarchy between them.
+3. **Tag or Digest**: This is the last part indicating the version of the image or artifact 
+you want to pull or push.
+
+#### References by Tag
+
+- **Tag**: Tags are user-friendly references to specific image or artifact versions.
+Common tags include `latest`, `v1.0`,
+or any other description that makes sense for versioning or identiying and resolve to a digest.
+These are typically human readable and easy to remember.
+
+![Diagram showing example of by tag reference](/img/concepts/reference/by_tag.png)
+
+:::caution
+Tags are mutable. This means that the same tag can point to different digests over time.
+:::
+
+#### References by Digest
+
+Reference by a digest is a unique, immutable identifier for a specific image or artifact.
+Typically represented as a SHA256 hash of the image's content, the digest guarantees
+you're pulling or referencing the exact version of the image and artifact and is **immutable**.
+
+![Diagram showing example of by digest reference](/img/concepts/reference/by_digest.png)
+
+:::note
+The digest is the most precise way to reference an image.
+However, it's not very user-friendly.
+This is why tags are more commonly used.
+A tag always resolves to a digest.
+:::
+
+#### References by Tag and Digest
+
+Putting it all together, a reference includes a registry, a repository, and a tag or a digest in the context
+of a **registry**.
+
+```mermaid
+graph LR 
+    A["Registry (e.g., docker.io)"]
+    B["Repository (e.g., library/nginx)"]
+    C["Tag (e.g., latest)"]
+    D["Digest (e.g., sha256:abc123...)"]
+    A --> B
+    B --> C
+    B --> D
+```
+
+### What a Reference Means in OCI Layout
+
+In the OCI Layout specification, a reference can be a tag or a digest.
```

Comment:
> a reference can be a tag or a digest.

To me it sounds a little bit weird since the location-addressable references should contain the directory path of the OCI layout. I checked image spec 1.1.0-rc4 and there is only place mentioning tag as a reference in the [index.json section](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/image-layout.md#indexjson-file), maybe it's not on-purpose and we can change the naming in the spec to `tag`?

To me below description of an OCI layout reference makes more sense: 

> In the OCI Layout, a reference is a combination of three parts:
> 1. **Path**: [an unique location identifier](https://en.wikipedia.org/wiki/Path_%28computing%29) to the layout directory
> 2. **Tag or Digest**: This is the last part indicating the version of the image or artifact stored in the layout.

Reply:
wrt index.json it’s only the tag. I think we could clarify that as a follow up.